### PR TITLE
Fix functional tests

### PR DESCRIPTION
**Pull request by eapolinario**
# TL;DR
Use restricted set of tests in functional tests

## Type
 - [ ] Bug Fix
 - [ ] Feature
 - [ ] Plugin

## Are all requirements met?

 - [ ] Code completed
 - [ ] Smoke tested
 - [ ] Unit tests added
 - [ ] Code documentation added
 - [ ] Any pending items have an associated Issue

## Complete description
Use a subset of tests on the demo cluster in functional tests.

## Tracking Issue
_Remove the '*fixes*' keyword if there will be multiple PRs to fix the linked issue_

fixes https://github.com/flyteorg/flyte/issues/<number>

## Follow-up issue
_NA_
OR
_https://github.com/flyteorg/flyte/issues/<number>_

### CODE REVIEW

**Comment by eapolinario** [comment]
This is the wrong approach. We need the flexibility of replacing components prior to running the tests. The way we do this now is by relying on the locally built component images and injecting them into the old sandbox as part of the process of running tests. 

The ultimate goal is to fully deprecate the old sandbox, but in order to do that we need the capability described in the previous paragraph. This is one more datapoint to expedite the move to a monorepo, where it'll be easy to setup these locally built single binary images containing updated code.

**Comment by eapolinario** [comment]
Functional tests passed: https://github.com/flyteorg/flyteadmin/actions/runs/5557184259/jobs/10150754879?pr=588. Will replicate this fix everywhere now.
